libbeat/dashboards/dashboards.go

```diff
@@ -26,19 +19,104 @@ func ImportDashboards(beatName, beatVersion string, esClient DashboardLoader, cf
 	dashConfig.URL = fmt.Sprintf(defaultURLPattern, beatVersion)
 	dashConfig.SnapshotURL = fmt.Sprintf(snapshotURLPattern, beatVersion)
 
-	err := cfg.Unpack(&dashConfig)
+	err := dashboardsConfig.Unpack(&dashConfig)
 	if err != nil {
 		return err
 	}
 
-	importer, err := NewImporter(&dashConfig, esClient, nil)
+	if esConfig != nil {
+		status, err := ImportDashboardsViaElasticsearch(esConfig, &dashConfig)
+		if err != nil {
+			return err
+		}
+		if status == true {
+			// the dashboards were imported via Elasticsearch
+			return nil
+		}
+	}
+
+	err = ImportDashboardsViaKibana(kibanaConfig, &dashConfig)
 	if err != nil {
-		return nil
+		return err
+	}
+
+	return nil
+}
+
+func ImportDashboardsViaKibana(config *common.Config, dashConfig *DashboardsConfig) error {
+
+	kibanaLoader, err := NewKibanaLoader(config, dashConfig, nil)
+	if err != nil {
+		return fmt.Errorf("fail to create the Kibana loader: %v", err)
+	}
+
+	if kibanaLoader == nil {
```

Comment:
Seems that etiher `err == nil` or `kibanaLoader != nil`, so I think the second check here is not needed and it doesn't follow our usual error checking pattern.